intra/doh/doh.go

```diff
@@ -74,7 +81,7 @@ type Listener interface {
 // so it has to be very simple.
 type Transport interface {
 	// Given a DNS query (including ID), returns a DNS response with matching
-	// ID, or an error if no response was received.
+	// ID, and/or an error if no response was received.
```

Comment:
Could you clarify under what conditions a non-nil error would be returned alongside a non-nil response? Is it during the hangover?

Reply:
Clarified.  It's whenever a SERVFAIL was synthesized.